pkg/controller/node/nodecontroller.go

```diff
@@ -334,13 +375,20 @@ func NewNodeController(
 						return
 					}
 				}
+				if nc.taintManager != nil {
+					nc.taintManager.NodeUpdated(node, nil)
+				}
 				if err := nc.cidrAllocator.ReleaseCIDR(node); err != nil {
 					glog.Errorf("Error releasing CIDR: %v", err)
 				}
 			},
 		}
 	}
 
+	if nc.runTaintManager {
+		nc.taintManager = NewNoExecuteTaintManager(kubeClient)
```

Comment:
There's a very weak link, that appears in the #41133 - namely `useTaintBasedEvictions:      useTaintBasedEvictions && runTaintManager`. Originally idea was that it'd be more tightly couples with the NC, but I managed to avoid that evenually, but the effect is that's a part of NC now.

It can easily be moved to completely separate controller (this is why I wrote it this way), but I'd rather avoid doing this change now - we can leave it as cleanup for 1.7.